src/Compilers/CSharp/Test/Emit2/Emit/RuntimeProbing/ModuleCancellationTests.cs

```diff
@@ -0,0 +1,2291 @@
+              IL_001a:  call       "void C<T>.G(T, System.Threading.CancellationToken)"
+              IL_001f:  nop
+              // sequence point: G(default, token);
+              IL_0020:  ldarg.0
+              IL_0021:  ldloca.s   V_0
+              IL_0023:  initobj    "T"
+              IL_0029:  ldloc.0
+              IL_002a:  ldsfld     "System.Threading.CancellationToken <PrivateImplementationDetails>.ModuleCancellationToken"
+              IL_002f:  call       "void C<T>.G(T, System.Threading.CancellationToken)"
+              IL_0034:  nop
+              // sequence point: }
+              IL_0035:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_GenericContainingType2()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C<S, T>
+            {
+                void F()
+                {
+                    G(default);
+                }
+
+                void G(S a)
+                {
+                }
+            
+                void G(T a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C<S, T>.F");
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_Generic_MatchingConstraint()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G<int>(1);
+                }
+
+                void G<T>(T a) where T : struct => throw null;
+                void G<T>(T a, CancellationToken token) where T : struct => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        verifier.VerifyMethodBody("C.F", """
+            {
+              // Code size       25 (0x19)
+              .maxstack  3
+              // sequence point: <hidden>
+              IL_0000:  ldsflda    "System.Threading.CancellationToken <PrivateImplementationDetails>.ModuleCancellationToken"
+              IL_0005:  call       "void System.Threading.CancellationToken.ThrowIfCancellationRequested()"
+              // sequence point: {
+              IL_000a:  nop
+              // sequence point: G<int>(1);
+              IL_000b:  ldarg.0
+              IL_000c:  ldc.i4.1
+              IL_000d:  ldsfld     "System.Threading.CancellationToken <PrivateImplementationDetails>.ModuleCancellationToken"
+              IL_0012:  call       "void C.G<int>(int, System.Threading.CancellationToken)"
+              IL_0017:  nop
+              // sequence point: }
+              IL_0018:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_Generic_MatchingConstraint_AlphaRename()
```

Comment:
>AlphaRename [](http://example.com/codeflow?start=72&length=11)

I think the use of the "AlphaRename" term is confusing here because it is only applicable to type parameters on the different form of the same symbol and really has nothing to do with names of type parameters themselves. In fact the names do not change during type parameter alpha rename performed by compiler.

Reply:
I see, I was using the term in the general sense of variable renaming that preserves semantics. Will rename.